src/js/apps/patients/worklist/worklist_state.js

```diff
@@ -195,8 +197,7 @@ export default Backbone.Model.extend({
     }, clone(currentSelectedList));
 
     this.set(listName, newSelectedList);
-
-    if (newLastSelectedId) this.set('lastSelectedId', newLastSelectedId);
+    this.set('lastSelectedId', newLastSelectedId || null);
```

Comment:
When the `selectMultiple()` function is called for the `click`+`shift` functionality, the `newLastSelectedId` parameter will be included. But when the function is used for the bulk select button (that previously used `selectAll()`), the `newLastSelectedId` function parameter will not be included.

Therefore, the `state.lastSelectedId` value will be set to `newLastSelectedId` if it exists as a function parameter. And set to `null` if not.

As a result, `state.lastSelectedId` will be set when the `click`+`shift` action is taken. And cleared when the user is using the bulk select button in UI.